bgpd/bgp_evpn.c

```diff
@@ -5661,6 +5661,10 @@ int bgp_evpn_local_l3vni_del(vni_t l3vni, vrf_id_t vrf_id)
 	for (ALL_LIST_ELEMENTS(bgp_vrf->l2vnis, node, next, vpn))
 		bgpevpn_unlink_from_l3vni(vpn);
 
+	if (CHECK_FLAG(bgp_vrf->vrf_flags, BGP_VRF_L3VNI_PREFIX_ROUTES_ONLY))
```

Comment:
Two questions:
1) This flag is SET in bgpevpn_link_to_l3vni() function. So it would be better to RESET this flag in bgpevpn_unlink_from_l3vni(). No functionality impact though. 
2) Do we need to check before resetting the flag ? Just reset the flag directly.

Reply:
@srimohans 
1) the flag is managed for l3vni creation/deletion with prefix-routes-only flag. It is not good to set the flag in bgpevpn_unlink_from_l3vni() as it is called from many other events. The sole purpose of this flag is to carry zebra configured 'prefix-routes-only' to bgp upon L3VNI creation and deletion.

2. I have submitted a new patch with just unsetting the flag when L3VNI is deleted under vrf from zebra.